### ISSUE TITLE
How to use weights for imbalanced datasets in Text Classification?

### ISSUE THREAD

**Issue by robinvanschaik** [flairNLP/flair#1897]
Hi all, 

This commit (#1345) added the option to include a dictionary with weights per class in order to help with imbalanced datasets. 
This can be seen as a [parameter](https://github.com/flairNLP/flair/pull/1345/files#diff-75c38d6f450d0363b795b61c783ba5f8R68) in the trainer function. 

However, this parameter is now [absent](https://github.com/flairNLP/flair/blob/master/flair/trainers/trainer.py#L63), but Iam guessing that the functionality might still be available?

In the meantime, I am using the ImbalancedClassificationDatasetSampler  as suggested [here](https://github.com/flairNLP/flair/issues/1094#issuecomment-538317563_), but apparently this is a bit experimental. 

In my case I am getting some solid improvements, but from what I hear the dict with class weights is a better approach. 

Is it still possible to pass a dict with class labels and weights in the trainer function?

**Comment by robinvanschaik** [comment]
After some digging I found out that the the weight dicts are no longer passed into the trainer, but into the classifier function.

For others looking into this, here is an example.

```python
import pandas as pd
import numpy as np
from sklearn.utils import class_weight

# Import modules from flair.
from flair.models import TextClassifier
from flair.trainers import ModelTrainer
from flair.data import Corpus
from flair.datasets import CSVClassificationCorpus
from torch.optim.adam import Adam
from flair.embeddings import SentenceTransformerDocumentEmbeddings
from flair.embeddings import TransformerDocumentEmbeddings
from flair.embeddings import WordEmbeddings, FlairEmbeddings, DocumentRNNEmbeddings

# Define the function for creating a weight dictionary.
def create_weight_dict(delimiter, label_index):

    training_dataset = pd.read_csv("./trainer/data/train.csv", delimiter = delimiter)

    unique_labels = np.unique(training_dataset.iloc[:, label_index])
    labels = training_dataset.iloc[:, label_index]

    class_weights = class_weight.compute_class_weight('balanced',
                                                      unique_labels,
                                                      labels)
    class_weights_dict = {}
    for unique_labels, class_weights in zip(unique_labels, class_weights):
                              class_weights_dict[unique_labels] = class_weights
    return class_weights_dict
```


Small end-to-end example how to incorporate in training flow.

```python
# 1. Column format indicating which columns hold the text and label(s)
column_name_map = {text_column_index: "text",
                       label_column_index: "label_topic"}

# 2. Load corpus containing training, test and dev data and if CSV has a header, you can skip it
corpus: Corpus = CSVClassificationCorpus("./trainer/data/",
                                             column_name_map,
                                             skip_header=True,
                                             delimiter=delimiter)

# Print statistics about the corpus.
training_data_statistics = corpus.obtain_statistics()
print(training_data_statistics)

# 3A. Create a label dictionary.
label_dict = corpus.make_label_dictionary()

# 3B. Calculate class weights.
weight_dict = create_weight_dict(delimiter=delimiter, 
label_index=label_column_index)

# 4. Use Transformer model.
document_embeddings = TransformerDocumentEmbeddings(
            model, fine_tune=True)

# 5. create the text classifier. 
    classifier = TextClassifier(
        document_embeddings, label_dictionary=label_dict, loss_weights=weight_dict)

# 6. initialize the text classifier trainer with Adam optimizer
trainer = ModelTrainer(classifier,
corpus,
optimizer=Adam,
use_tensorboard=False)

# 7. start the training
trainer.train("./trainer/checkpoint/",
                  learning_rate=3e-5,  # use very small learning rate
                  max_epochs=max_epochs,
                  patience=patience,
                  checkpoint=True,
                  mini_batch_size=16)

```
